app/routes/people/person/personal-information.js

```diff
@@ -11,11 +17,99 @@ export default class PeoplePersonPersonalInformationRoute extends Route {
         'mandatories.contacts',
+        'agents-in-position',
       ].join(),
     });
+    const contacts = [];
+    const m = await person.mandatories;
+    console.log(m);
+    const mandatories = person.mandatories.filter(
+      (m) => m != null // hack because otherwise mandatories is not an array
+    );
+
+    const ministers = person.agentsInPosition.filter(
+      (m) => m != null // same
+    );
+
+    for (let mandatory of mandatories) {
+      const mandate = await mandatory.mandate;
+      if (!this.isActivePosition(mandate.endDate)) {
+        break;
+      }
+      const role = await mandate.roleBoard;
+      const governingBody = await mandate.governingBody;
+      const isTimeSpecializationOf = await governingBody.isTimeSpecializationOf;
+      const administrativeUnit =
+        await isTimeSpecializationOf.administrativeUnit;
+      const mContacts = await mandatory.contacts;
+      const primaryContact = findPrimaryContact(mContacts);
+      const secondaryContact = findSecondaryContact(mContacts);
+      contacts.push({
+        title: `${role.label}, ${administrativeUnit.name}`,
+        role: role.label,
+        type: 'mandatory',
+        id: mandatory.id,
+        startDate: mandate.startDate,
+        endDate: mandate.endDate,
+        unitId: administrativeUnit.id,
+        unitName: administrativeUnit.name,
+        primaryContact: await this.mapContact(primaryContact),
+        secondaryContact: await this.mapContact(secondaryContact),
+      });
+    }
 
+    for (let minister of ministers) {
+      if (!this.isActivePosition(minister.agentEndDate)) {
+        break;
+      }
+      const position = await minister.position;
+      const role = await position.function;
+      const administrativeUnit = await position.worshipService;
+      const mContacts = await minister.contacts;
+      const primaryContact = findPrimaryContact(mContacts);
+      const secondaryContact = findSecondaryContact(mContacts);
+      contacts.push({
+        title: `${role.label}, ${administrativeUnit.name}`,
+        role: role.label,
+        type: 'minister',
+        id: minister.id,
+        startDate: minister.agentStartDate,
+        endDate: minister.agentEndDate,
+        unitId: administrativeUnit.id,
+        unitName: administrativeUnit.name,
+        primaryContact: await this.mapContact(primaryContact),
+        secondaryContact: await this.mapContact(secondaryContact),
+      });
+    }
     return {
       person,
+      contacts: contacts.sort((a, b) => {
+        return new Date(b.startDate) - new Date(a.startDate);
+      }),
     };
   }
+  async mapContact(contact) {
+    if (!contact) return null;
+    return {
+      email: contact.email,
+      telephone: contact.telephone,
+      fax: contact.fax,
+      website: contact.website,
+      contactAddress: await contact.contactAddress,
+    };
+  }
+  isActivePosition(endDate) {
```

Comment:
I would move this into a util function that we can use both here and in the status component.